### DISCUSSION TITLE
Remove object from a ui.scene()

### DISCUSSION THREAD

**Discussion by Bardreamaster** [zauberzeug/nicegui#803] [Q&A] [answered]
### Question

I'm going to have a cube view scene by using `ui.scene()`. While each time user click a button with number `n` in a `number` input, there should be `n scene.box()` shown in the scene.  But I only find how to add new boxes in but no delete/remove. So if I want to refresh the viewer, I have to remove the `ui.scene` in a row container, and regenerate it again to add different number objects in.

It feels like a little weird to regenerate an element but I only want refresh the objects in `ui.scene()`. Is there any way to remove/delete a object in scene or just clear all object in a scene?

Thanks so much.

**Comment by falkoschindler** [comment, marked as answer]
Hi @Bardreamaster!
Every 3D object has a `delete` method which you can use to remove individual objects from a scene:

```py
with ui.scene() as scene:
    sphere = scene.sphere()

ui.button('Delete sphere', on_click=sphere.delete)
```

**Reply by Bardreamaster** [reply]
@falkoschindler  Thanks so much! And there the way I generate many objects and refresh them all:

```python
from nicegui import ui

def clear():
    #delete every box in the scene
    for box in box_list:
        box.delete()
   # clear the list to receive new objects to avoid exception by deleting non-existent object
    box_list.clear()

def generate():
    with scene :
        # add any object in the scene and also in the list
        for n in range(5):
            box_list.append(scene.box().move(x=n))

box_list = []
with ui.row():
    with ui.scene(width=960, height=720) as scene:
       # Initial a scene first
        with scene.group().move(z=2) as group:
            scene.box().move(x=2)
            scene.box().move(y=2).rotate(0.25, 0.5, 0.75)
            scene.box(wireframe=True).material('#888888').move(x=2, y=2)
```


And if this `clear/remove/delete` is available in ui.scene() with `add/insert` method, it would be better I think. Maybe there are reasons not to do so.

Again, thanks a lot!

**Reply by falkoschindler** [reply]
@Bardreamaster Why would awaitable methods be better? NiceGUI's update methods usually just put a message in an outbox which is processed in the background. This performs much better than awaiting each individual message being sent. And processing the outbox in a single background process saves the creation of a new background task for every message.
